middleware/http/wasm/internal/test/logger.go

```diff
@@ -0,0 +1,81 @@
+package test
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/dapr/kit/logger"
+)
+
+// compile-time check to ensure testLogger implements logger.testLogger.
+var _ logger.Logger = &testLogger{}
+
+func NewLogger() logger.Logger {
```

Comment:
Why this logger?

Reply:
for benchmarks, so that they don't end up using a real one

Reply:
@ItalyPaleAle sorry I added a commit, I left the old copy. The benchmarks are in the root directory of wasm as they will be re-used when we add the flexible wasm thing next (ex compare existing basic vs the new http-wasm one)

Reply:
Dapr/kit/logger now supports setting a buffer as log destination. Would that work? (Just fetch the latest version from main)
I am just worried that this logger will need to be updated too every time the log interface changes.

Reply:
yeah I can refactor it out. I'll try it!

Reply:
I think the other thing was testing was more difficult due to the amount of formatting logic internal to it. we'll see

Reply:
It looks like this isn't exported @ItalyPaleAle 

I can see where some code sets does this, but it is within package. Am I missing something?

```
func getTestLogger(buf io.Writer) *daprLogger {
	l := newDaprLogger(fakeLoggerName)
	l.logger.Logger.SetOutput(buf)

	return l
}
```

Reply:
in any case, the logger here was moved not written from scratch. can we decouple this PR from making the upstream logger capable of being used in test mode maybe?

Reply:
ah I see you added here.. if it is ok to use a non-tagged version I can do that in this PR. otherwise we need a tag on that repo https://github.com/dapr/kit/pull/25

Reply:
Using a non-tagged version is fine

Reply:
hmm this isn't working out well, as something in the path of the request is setting the internal level to panic. I've not hunted down what, yet.

Reply:
I got it. I'll use a master commit for dapr/kit for now.